fpm/src/fpm_sources.f90

```diff
@@ -0,0 +1,504 @@
+module fpm_sources
+use fpm_filesystem, only: basename, read_lines, list_files
+use fpm_strings, only: lower, split, str_ends_with, string_t
+use fpm_manifest_executable, only: executable_t
+implicit none
+
+private
+public :: srcfile_ptr, srcfile_t
+public :: add_sources_from_dir, add_executable_sources, resolve_module_dependencies
+public :: FPM_UNIT_UNKNOWN, FPM_UNIT_PROGRAM, FPM_UNIT_MODULE, &
+          FPM_UNIT_SUBMODULE, FPM_UNIT_SUBPROGRAM, FPM_UNIT_CSOURCE, &
+          FPM_UNIT_CHEADER
+
+integer, parameter :: FPM_UNIT_UNKNOWN = -1
+integer, parameter :: FPM_UNIT_PROGRAM = 1
+integer, parameter :: FPM_UNIT_MODULE = 2
+integer, parameter :: FPM_UNIT_SUBMODULE = 3
+integer, parameter :: FPM_UNIT_SUBPROGRAM = 4
+integer, parameter :: FPM_UNIT_CSOURCE = 5
+integer, parameter :: FPM_UNIT_CHEADER = 6
+
+character(15), parameter :: INTRINSIC_MODULE_NAMES(*) =  &
+                             ['iso_c_binding  ', &
+                              'iso_fortran_env']
```

Comment:
What about the `ieee_arithmetic`, `ieee_exceptions` and `ieee_features` modules? See https://gcc.gnu.org/onlinedocs/gfortran/IEEE-modules.html